project-plan.md

```diff
@@ -24,50 +24,95 @@ for programming because they facilitate exploratory programming and debugging.
 Common examples include command-line shells such as Bash and Python's REPL.
 
 The deliverable for this project, then, is to create such a REPL generator for
-the Spoofax Language Workbench. This REPL should provide the following features:
-
-* TODO: based on the research report and the minutes of meeting 2016-04-19,
-  make our own list of features that we will propose to Eelco and Gabriel.
-
-Non-features (e.g. features that should *not* be implemented) are:
-
-* TODO: see above.
+the Spoofax Language Workbench.
+
+# The final product
+The final product will meet the following requirements and demands,
+as specified following the MoSCoW method:
+
+* Must-have
+	* Interactive REPL
+		* Generated from language definition
+		* Should be language definition agnostic
+		* Syntax checked expressions
+	* Indication of errors
+	* Integration with Eclipse
+	* Syntax highlighting
+* Should-have
+	* Save and load shell state
+* Could-have
+	* Integration with other IDE's (IntelliJ)
+	* Hover over variables to see value, type and others
+	* GDB style interaction with context / environment
+		* Ability to partially change or update environment
+	* Literate programming
+* Won't-have
 
 If the above turns out to be (too) easy, the REPL can be extended into a
 language playground such as the one offered by the Swift programming language.
 If we decide to extend the project, new agreements will be made between us and
 the client.
 
-## Methodologies and tooling
+# Methodologies and tooling
 
 During this project we will work using the SCRUM methodology with weekly
 sprints. Our backlog, current and completed sprints will be managed using
-Trello. Trello will also be used to communicate documents to the client
-and the TU coach.
-
-We will use pull-based development using GitHub. GitHub Issues will be used to
-track issues (all of this will be linked to Trello in one way or another). To
-facilitate this, we will be given our own repository within the Metaborg
-organization. Each of us will then fork this repository to our private GitHub
-accounts. We will also use one private repository to hold all the documents
+[Trello](https://trello.com/b/u2aKQ12y/bachelor-project-spoofax-repl).
+Trello will also be used to communicate documents to the client and the TU coach.
+Sprints will end on friday, coinciding with most deadlines according to the
+planning for this project as mandated by TU Delft.
+
+Each sprint will start with a "sprint planning event", in which user stories
+from the backlog will be assigned to project members, based on their
+prioritization and their feasibility for this sprint.
+Each sprint will end with a meeting in which we discuss to what degree
+we have achieved our sprint planning and how we can improve on that.
```

Comment:
A summary of these meetings will be discussed with Hendrik on the following Monday during our weekly meetings?

I think we need to think how we can combine these sessions with the weekly meetings.